src/Migrations/SyncUserToPTUser/SyncUserToPTUserService.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.ExceptionServices;
+using System.Threading.Tasks;
+using Autofac;
+using MongoDB.Driver;
+using SIL.XForge.DataAccess;
+using SIL.XForge.Models;
+using SIL.XForge.Realtime;
+using SIL.XForge.Realtime.Json0;
+using SIL.XForge.Scripture.Models;
+using SIL.XForge.Scripture.Services;
+
+namespace SyncUserToPTUser
+{
+    /// <summary> Code to move sync users to the sf-project model. </summary>
+    public class SyncUserToPTUserService : ISyncUserToPTUserService
+    {
+        private readonly IRealtimeService _realtimeService;
+        private readonly IParatextService _paratextService;
+        private readonly IRepository<SFProjectSecret> _projectSecretRepo;
+        private IConnection _realtimeServiceConnection;
+
+        private readonly IProgramLogger _logger;
+
+        public SyncUserToPTUserService(IRealtimeService realtimeService, IParatextService paratextService,
+            IRepository<SFProjectSecret> projectSecretRepo, IProgramLogger logger)
+        {
+            _realtimeService = realtimeService;
+            _paratextService = paratextService;
+            _projectSecretRepo = projectSecretRepo;
+            _logger = logger;
+        }
+
+        /// <summary> Move sync users to the SF project model. </summary>
+        public async Task MoveSyncUsersToProject(bool dryRun, ISet<string> sfProjectIdsToUpdate = null)
+        {
+            List<SFProject> sfProjects = _realtimeService.QuerySnapshots<SFProject>().ToList<SFProject>();
+            if (sfProjectIdsToUpdate != null)
+            {
+                sfProjects.RemoveAll((SFProject sfProject) => !sfProjectIdsToUpdate.Contains(sfProject.Id));
```

Comment:
Might be actual problem:

Does RemoveAll() not need to return something that we keep? (Rather than modify the set in place). 

Just saying, but not needed:

It might be a bit better conceptually to look for what you want to keep, rather than remove what you don't want to keep. You might do something like
```suggestion
                sfProjects = sfProjects.Where((SFProject sfProject) => sfProjectIdsToUpdate.Contains(sfProject.Id));
```

In fact, you could even start off the method with something like the following:
```cs
if (sfProjectIdsToUpdate == null) {
  throw new ArgumentNullException(nameof(sfProjectIdsToUpdate));
}
List<SFProject> sfProjects = _realtimeService.QuerySnapshots<SFProject>()
  .ToList<SFProject>()
  .Where((SFProject sfProject) => sfProjectIdsToUpdate.Contains(sfProject.Id));
```